dpe/src/commands/certify_key.rs

```diff
@@ -19,21 +19,48 @@ pub struct CertifyKeyCmd {
     pub label: [u8; DPE_PROFILE.get_hash_size()],
 }
 
+impl CertifyKeyCmd {
+    const _ND_DERIVATION: u32 = 1 << 31;
+
+    // Uses non-deterministic derivation.
+    const fn _uses_nd_derivation(&self) -> bool {
+        self.flags & Self::_ND_DERIVATION != 0
+    }
+}
+
 impl<C: Crypto> CommandExecution<C> for CertifyKeyCmd {
     fn execute(&self, dpe: &mut DpeInstance<C>, locality: u32) -> Result<Response, DpeErrorCode> {
+        // Make sure the operation is supported.
+        if !dpe.support.nd_derivation && self._uses_nd_derivation() {
+            return Err(DpeErrorCode::InvalidArgument);
+        }
+
         let idx = dpe
             .get_active_context_pos(&self.handle, locality)
             .ok_or(DpeErrorCode::InvalidHandle)?;
+        let context = &dpe.contexts[idx];
 
         // Make sure the command is coming from the right locality.
-        if dpe.contexts[idx].locality != locality {
+        if context.locality != locality {
             return Err(DpeErrorCode::InvalidHandle);
         }
 
-        // Derive CDI and public key
-        let cdi = dpe.derive_cdi(idx)?;
-        let pub_key = C::derive_ecdsa_pub(DPE_PROFILE.alg_len(), &cdi, &self.label, b"ECC")
-            .map_err(|_| DpeErrorCode::InternalError)?;
+        let algs = DPE_PROFILE.alg_len();
+        let pub_key = if context.cached_priv_key.is_some() {
```

Comment:
Should we only take the cached key if this request is also non-deterministic? In other words

```Rust
if self.uses_nd_derivation && context.cached_priv_key.is_some()
```